### PR TITLE
add cambridge latex style

### PR DESCRIPTION
**Pull request by cheungdaven**
add Cambridge latex style. 
We can use the following command to invoke the Cambridge latex:
d2lbook build pdf --style cambridge

### CODE REVIEW

**Comment by cheungdaven** [comment]
The latex function is ready for review. @astonzhang

File: d2lbook/sphinx_template.py [hunk 1]
```diff
@@ -52,8 +52,6 @@
 
 rsvg_converter_args = ['-z', '0.8']
 
-bibtex_bibfiles = ["BIBFILE"]
```
Comment (astonzhang) [inline d2lbook/sphinx_template.py:55]: Why do we need to remove this under `sphinx_conf`?

File: d2lbook/build.py [hunk 2]
```diff
@@ -33,23 +36,31 @@ def build():
     parser.add_argument('commands', nargs='+', choices=commands)
     parser.add_argument('--tab', default=None,
                         help='The tab to build, if multi-tab is enabled.')
+    parser.add_argument('--style', default=None,
+                        help='The pdf style. It supports cambridge style and sphinx default.')
```
Comment (astonzhang) [inline d2lbook/build.py:40]:
```suggestion
                        help='The pdf style. It supports cambridge style (cambridge) and sphinx default (None).')
```

File: d2lbook/sphinx_template.py (hunk 1 above)
Comment (cheungdaven) [reply d2lbook/sphinx_template.py:55]: It might be a sync issue.

Comment (astonzhang) [reply d2lbook/sphinx_template.py:55]: I'm adding it back per https://github.com/d2l-ai/d2l-book/commit/89db4fed7351ee289f63ea60f8baee7d1fd1cd87

File: d2lbook/sphinx_template.py [hunk 3]
```diff
@@ -52,8 +52,6 @@
 
 rsvg_converter_args = ['-z', '0.8']
```
Comment (astonzhang) [inline d2lbook/sphinx_template.py:54]:
```suggestion
bibtex_bibfiles = ["BIBFILE"]
```

File: d2lbook/sphinx_template.py [hunk 4]
```diff
@@ -116,6 +114,186 @@ def setup(app):
...
+
+latex_documents = [
+    (master_doc, "NAME.tex", "TITLE",
+     author, 'PT1'),
+]
+
+
+
+rsvg_converter_args = ['-z', '0.8']
+
```
Comment (astonzhang) [inline d2lbook/sphinx_template.py:171]:
```suggestion

bibtex_bibfiles = ["BIBFILE"]

```

File: d2lbook/build.py [hunk 5]
```diff
@@ -33,23 +36,31 @@ def build():
...
                         help='The tab to build, if multi-tab is enabled.')
+    parser.add_argument('--style', default=None,
+                        help='The pdf style. It supports cambridge style (cambridge) and sphinx default (None).')
     args = parser.parse_args(sys.argv[2:])
     config = Config(tab=args.tab)
     builder = Builder(config)
     for cmd in args.commands:
-        getattr(builder, cmd)()
+        if cmd == 'pdf':
+            getattr(builder, cmd)(args.style)
```
Comment (astonzhang) [inline d2lbook/build.py:46]: Can you pass the `style` flag similar to `config = Config(tab=args.tab)`? In this case every member method can access it via `self.config.style` without passing via method arguments.

**Comment by astonzhang** [comment]
Besides, please sync changes (e.g., different background colors for input code cells and output code cells) from the master branch with updated environment (e.g., `Sphinx 5.1.1` and `sphinxcontrib-bibtex 2.4.2`).

File: d2lbook/build.py [hunk 6]
```diff
@@ -356,6 +388,9 @@ def linkcheck(self):
     @_once
     def pdf(self):
         self.rst()
+        if self.config.pdf['style']  is not None:
```
Comment (astonzhang) [inline d2lbook/build.py:391]: not equal to `'Default'`?

Comment (cheungdaven) [reply d2lbook/build.py]: changed to == 'cambridge'

File: d2lbook/build.py (hunk 5 above)
Comment (cheungdaven) [reply d2lbook/build.py]: done. style argument is removed.

File: d2lbook/config_default.ini [hunk 7]
```diff
@@ -136,6 +136,11 @@ latex_logo =
 # Bibtext bibfile
 bibfile =
 
+# Specify the url of external latex resources.
+latex_url = 
+
+# Specify the latex style. We now support "cambridge" and the defaut sphinx style.
+style =
```
Comment (astonzhang) [inline d2lbook/config_default.ini:143]: Can you set the default value here？So users don't need to specify style = Default if they use default style.

Comment (cheungdaven) [reply d2lbook/config_default.ini:143]: they actually do not need to put a default there. it can be empty.

File: d2lbook/sphinx.py [hunk 8]
```diff
@@ -5,14 +5,17 @@
 
 __all__ = ['prepare_sphinx_env']
 
-def prepare_sphinx_env(config):
-    env = SphinxEnv(config)
+def prepare_sphinx_env(config, style):
```
Comment (astonzhang) [inline d2lbook/sphinx.py:8]: remove `style` argument? See https://github.com/d2l-ai/d2l-book/blob/cc9d457bf75a1056021aa7ea01de99d656dc4ef9/d2lbook/build.py#L327

File: d2lbook/sphinx.py [hunk 9]
```diff
@@ -5,14 +5,17 @@
...
+    env = SphinxEnv(config, style)
```
Comment (astonzhang) [inline d2lbook/sphinx.py:9]: remove `style` argument

Comment (cheungdaven) [reply d2lbook/sphinx.py]: done.

File: d2lbook/config_default.ini (hunk 7 above)
Comment (astonzhang) [reply d2lbook/config_default.ini:143]: Agreed that leaving it empty without any arbitrary default string is better

**Comment by astonzhang** [comment]
Thanks for adding this new feature!

**Pull request merged**